taca_ngi_pipeline/deliver/deliver.py

```diff
@@ -87,19 +87,18 @@ def __init__(self, projectid, sampleid, **kwargs):
         self.reportpath = getattr(self, 'reportpath', None)
         self.force = getattr(self, 'force', False)
         self.stage_only = getattr(self, 'stage_only', False)
+        self.projectname = None
         #Fetches a project name, should always be availble; but is not a requirement
         try:
-            n = db.project_entry(db.dbcon(), projectid)['name']
-            self.projectname = n
+            self.projectname = getattr(db.project_entry(db.dbcon(), projectid), 'name')
         except KeyError:
             pass
         # only set an attribute for uppnexid if it's actually given or in the db
         try:
             getattr(self, 'uppnexid')
         except AttributeError:
             try:
-                t = db.project_entry(db.dbcon(), projectid)['uppnex_id']
-                self.uppnexid = t
+                self.uppnexid = getattr(db.project_entry(db.dbcon(), projectid), 'uppnexid', None)
```

Comment:
Actually, the quite ugly construct here was done with the intention to only set the `uppnexid` attribute if we really, really have been given a value for it, i.e. setting it to `None` was not good enough. The reason for this is that the `expand_path` method does not handle a `None` attribute. This is neither very clear nor elegant and should probably be rewritten but if you want to keep on the safe side, better to leave this code the way it was.